GRCCA: Checking on feature groups fails for feature groups with np.int8

In [`cca_zoo.models.GRCCA`](https://github.com/jameschapman19/cca_zoo/blob/7575d3b47b8ee2c62a8f004141d09bf23055ae83/cca_zoo/models/_grcca.py#L58-L59) each feature group is checked for an integer data type. This however fails for `np.int8`:

Example:

```python
import numpy as np
from cca_zoo.models import GRCCA

# create two random matrices and pretend both of them would have two feature
# groups
rng = np.random.RandomState(0)
X1 = rng.random((100,4))
X2 = rng.random((100,4))
feature_groups = [np.array([0,0,1,1],dtype=np.int8),np.array([0,0,-1,-1])]

# fit GRCCA on the data
estimator = GRCCA()
estimator.fit([X1,X2],feature_groups=feature_groups)


  File "/home/johannes.wiesner/.conda/envs/csp_wiesner_johannes/lib/python3.9/site-packages/cca_zoo/models/_grcca.py", line 59, in fit
    assert feature_group.dtype == int, "subject_groups must be integers"

AssertionError: subject_groups must be integers

```

I ran into this error because I converted a `pandas.Series` to an integer array. This could a typical workflow because you might start with X1 and X2 being multi-index `pd.DataFrames` (where the first column level describes the variable names and the second level describes groups of these variables)

```python
import pandas as pd
feature_group = pd.Series(['cognition','cognition','psychopathology','psychopathology']).astype('category').cat.codes.values
feature_group.dtype
Out[32]: dtype('int8')
```

Solution:

Use a more generic check like:

```python
for feature_group in feature_groups:
    assert np.issubdtype(feature_group.dtype,np.integer), "feature groups must be integers"
```

Also: change the error message from "subject groups must be integers"  to "**feature** groups must be integers"

I will submit a pull-request :)

--- a/cca_zoo/models/_grcca.py
+++ b/cca_zoo/models/_grcca.py
@@ -56,7 +56,7 @@ class GRCCA(MCCA):
             warnings.warn(f"No feature groups provided, using all features")
             feature_groups = [np.ones(view.shape[1], dtype=int) for view in views]
         for feature_group in feature_groups:
-            assert feature_group.dtype == int, "subject_groups must be integers"
+            assert np.issubdtype(feature_group.dtype, np.integer), "feature groups must be integers"
         views = self._validate_inputs(views)
         self._check_params()
         views, idxs = self._preprocess(views, feature_groups)

